digdag-core/src/main/java/io/digdag/core/repository/Revision.java

```diff
@@ -19,6 +19,8 @@
 
     public abstract Optional<String> getArchivePath();
 
+    public abstract Optional<Config> getUserInfo();
```

Comment:
I think it's better to use `Config` instead of `Optional<Config>` because difference of `null` from `{}` is not always clear. Moreover, when `io.digdag.core.database.ConfigMapper` gets data from a text column, it doesn't distinguish null and `{}` (both of them become `{}`).

AuthRequestFilter can inject ConfigFactory at its constructor so that setProperty can replace absent to `{}` like `requestContext.setProperty("userInfo", result.getUserInfo().or(cf.create()))`